wine_utils/dlls_setup.sh.in

```diff
@@ -14,59 +18,48 @@ if [ -z "$WINEPREFIX" ]; then
-    cleared_path=$(grep REG_EXPAND_SZ <<< $path | sed -E -e $path_pattern -e "s|^(.*;)?$dlls_windir_pattern(;(.*))?$|\\1\\3|")
-    echo "$path"
-    echo "new PATH:                    $cleared_path"
-    wine reg add 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' /v PATH /d "$cleared_path" /t REG_EXPAND_SZ
-    exit 0
-fi
-
-function redirect {
-    wine reg query HKEY_CURRENT_USER\\Software\\Wine\\DllRedirects /v $2 | grep -E '^\s*'"$2"'\s+REG_SZ\s+'"$2"'_vk.dll\s*$' 1>/dev/null
-    if [ $? -eq 1 ]; then
-        echo "redirecting [$2 -> $2_vk]"
-        wine reg add HKEY_CURRENT_USER\\Software\\Wine\\DllRedirects /v $2 /d $2_vk.dll
+function removeOverride {
+    echo "$1:"
+    echo -n '    [1/2] Removing override... '
+    wine reg add 'HKEY_CURRENT_USER\Software\Wine\DllOverrides' /v $1 /d builtin /f
+    if [ ! $? ]; then
+        exit 1
+    fi
+    local dll="$unix_sys_path/$1.dll"
+    echo -n '    [2/2] Removing link... '
+    if [ -h "$dll" ]; then
+        rm "$dll"
+        if [ "$?" == "0" ]; then
+            echo "Done."
+        fi
     else
-        echo -e "redirection [$2 -> $2_vk] was \\e[0;32mOK\\e[0m"
+        echo "'$dll' is not a link or doesn't exist."
     fi
 }
 
-function addToPATH {
-    path=$(wine reg query 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' /v PATH | sed -e 's|\r||g')
-    path_val=$(grep REG_EXPAND_SZ<<<$path | sed -E -e $path_pattern)
-    check=$(sed -E -e "s|^(.*;)?$(sed 's|\\|\\\\|g'<<<$1)(;.*)?$||" <<<$path_val)
-    if [ -n "$check" ]; then
-        echo "adding $1 to windows PATH"
-        new_path="$(sed -E  -e 's|;?\s*$||' <<<$path_val);$1"
-        echo "$path"
-        echo "new PATH:                    $new_path"
-        wine reg add 'HKEY_LOCAL_MACHINE\System\CurrentControlSet\Control\Session Manager\Environment' /v PATH /d "$new_path" /t REG_EXPAND_SZ
-    else
-        echo -e "windows PATH was \\e[0;32mOK\\e[0m"
+function createOverride {
+    echo "$1:"
+    echo -n '    [1/2] Creating override... '
+    wine reg add 'HKEY_CURRENT_USER\Software\Wine\DllOverrides' /v $1 /d native /f
+    echo -n "    [2/2] Creating link to $1.dll... "
+    ln -sf "$dlls_dir/$1.dll" "$unix_sys_path/$1.dll"
+    if [ $? ]; then
+        echo "Done."
     fi
 }
 
-if [ "$1" == "overrides" ]; then
-    set -e
-    if [ $build_arch == "x86_64" ]; then
-        wine=wine64
-    else
-        wine=wine
-    fi
-    unix_sys_path="$($wine winepath -u 'C:\windows\system32')"
-    wine reg add HKEY_CURRENT_USER\\Software\\Wine\\DllOverrides /v d3d11 /d native /f
-    wine reg add HKEY_CURRENT_USER\\Software\\Wine\\DllOverrides /v dxgi /d native /f
-    ln -sf "$dlls_dir/d3d11_vk.dll" "$unix_sys_path/d3d11.dll"
-    ln -sf "$dlls_dir/dxgi_vk.dll" "$unix_sys_path/dxgi.dll"
-    exit 0
+if [ "$1" == "reset" ]; then
+    echo -n '[1/2] '
+    removeOverride d3d11
+    echo -n '[2/2] '
+    removeOverride dxgi
+    exit
 fi
```

Comment:
Is there any particular reason you're not simply using an `else` here instead of that `exit; fi` construct? This makes the script harder to read in my opinion.

Reply:
Nothing really, just a stylistic choice. It doesn't feel natural to have the main code in a `else` statement. This way it's clearer that the reset option is the alt case, and the other is the main purpose of the script.